python/tank/bootstrap/cached_configuration.py

```diff
@@ -25,6 +27,15 @@
 
 log = LogManager.get_logger(__name__)
 
+def _get_perm(fname):
+    return stat.S_IMODE(os.lstat(fname)[stat.ST_MODE])
+
+def _make_writeable_recursive(path):
+    for root, dirs, files in os.walk(path, topdown=False):
+        for dir in [os.path.join(root, d) for d in dirs]:
```

Comment:
I've some helper functions to the sgtk.util.filesystem files based on your recommendation. I have NOT added the unit tests for them because of lack of time. I will do that when i get back, if you are OK with the API I've defined